flink-connector-jdbc-postgres/src/main/java/org/apache/flink/connector/jdbc/postgres/database/dialect/CompatiblePostgresDialectConverter.java

```diff
@@ -18,11 +18,11 @@
 
 package org.apache.flink.connector.jdbc.postgres.database.dialect;
 
-import org.apache.flink.annotation.PublicEvolving;
+import org.apache.flink.annotation.Internal;
 import org.apache.flink.table.types.logical.RowType;
 
 /** JDBC converter for PostgreSQL compatible databases. */
-@PublicEvolving
+@Internal
```

Comment:
this is unclear to me
shouldn't we create a kind of FLIP first if we make such changes?

Reply:
This configuration is currently misaligned. All dialect implementations are classified as internal. The purpose of this class is solely to allow the CrateDB dialect to utilize the PostgreSQL implementation.

To preserve the PublicEvolving status, we need to add an exception to the architecture test. Without this exception, the rule "Public Evolving API Methods Use Only Public Evolving API Types" will fail.

To resolve this issue without an exception, we would need to refactor all dialect implementations and classify them as PublicEvolving.

I can add the exception if you still believe that a FLIP is necessary and focus the discussion solely on that change. I believe this will not impact the rest of the PR.

Reply:
>I can add the exception if you still believe that a FLIP is necessary and focus the discussion solely on that change. I believe this will not impact the rest of the PR.

would be great.
Agree, rest of the PR is more or less ok (I did only a brief look, will look again a bit later)